merlin/systems/dag/ops/operator.py

```diff
@@ -41,22 +51,97 @@ def __repr__(self):
 
 
 class InferenceOperator(BaseOperator):
+    """
+    The basic building block of the Inference Node, the Inference Operator encapsulates
+    a transform with the appropriate functionality to handle all Ensemble graph
+    operations.
+    """
+
     @property
     def export_name(self):
+        """
+        Provides a clear common english identifier for this operator.
+
+        Returns
+        -------
+        String
+            Name of the current class as spelled in module.
+        """
         return self.__class__.__name__.lower()
 
     @abstractmethod
-    def export(self, export_path, input_schema, output_schema, node_id=None, version=1):
-        pass
+    def export(
+        self,
+        path: str,
+        input_schema: Schema,
+        output_schema: Schema,
+        params: dict = None,
+        node_id: int = None,
+        version: int = 1,
+    ) -> tuple(dict, list):
+        """
+        Export the class object as a config and all related files to the user defined path.
+
+        Parameters
+        ----------
+        path : str
+            Artifact export path
+        input_schema : Schema
+            A schema with information about the inputs to this operator.
+        output_schema : Schema
+            A schema with information about the outputs of this operator.
+        params : dict, optional
+            Parameters dictionary of key, value pairs stored in exported config, by default None.
+        node_id : int, optional
+            The placement of the node in the graph (starts at 1), by default None.
+        version : int, optional
+            The version of the model, by default 1.
+
+        Returns
+        -------
+        Ensemble_config: dict
+            The config for the entire ensemble.
+        Node_configs: list
+            A list of individual configs for each step (operator) in graph.
+        """
+
+    def create_node(self, selector: ColumnSelector) -> InferenceNode:
+        """_summary_
+
+        Parameters
+        ----------
+        selector : ColumnSelector
+            Selector to turn into an Inference Node.
 
-    def create_node(self, selector):
+        Returns
+        -------
+        InferenceNode
+            New node for Ensemble graph.
+        """
         return InferenceNode(selector)
 
 
 class PipelineableInferenceOperator(InferenceOperator):
+    """
+    This Inference operator type builds on the base infrence operator, by allowing
```

Comment:
My $0.02 is to lowercase the first "Inference".  The rest reads perfectly clearly without inline-code styling.

s/infrence/inference/

In the last sentence, maybe "This ability removes overhead from Triton Inference Server between operators of this type." (Might not be perfect.)